api/v1alpha1/jvmchaos_types.go

```diff
@@ -0,0 +1,299 @@
+	// +optional
+	File string `json:"file"`
+
+	// Name represents script name, use for logging
+	// +optional
+	Name string `json:"name"`
+
+	// Type represents script type, java or groovy, default to java
+	// +kubebuilder:validation:Enum=java;groovy;
+	// +optional
+	Type string `json:"type"`
+}
+
+// JVMReturnSpec represents the detail about JVM chaos action of return value
+type JVMReturnSpec struct {
+	// JVMCommonParameter represents the common jvm chaos parameter
+	JVMCommonParameter `json:",inline"`
+
+	// Value represents specifies the return value of a class method, supporting only primitive, null, and String return values. required
+	Value string `json:"value"`
+}
+
+// JVMDelaySpec represents the detail about JVM chaos action of invoke delay
+type JVMDelaySpec struct {
+	// JVMCommonParameter represents the common jvm chaos parameter
+	JVMCommonParameter `json:",inline"`
+
+	// Time represents delay time, in milliseconds, required
+	Time int `json:"time"`
+
+	// Offset represents delay fluctuation time
+	// +optional
+	Offset int `json:"offset"`
+}
+
+// JVMCommonParameter represents the common jvm chaos parameter
+type JVMCommonParameter struct {
+	// Classname represents specify the class name, which must be an implementation class with a full package name, such as com.xxx.xxx.XController. required
+	Classname string `json:"classname"`
+
+	// Methodname represents specify the method name. Note that methods with the same method name will be injected with the same fault. required
+	Methodname string `json:"methodname"`
+
+	// After represents method execution is completed before the injection failure is returned.
+	// +optional
+	After bool `json:"after"`
+}
+
+// ServletExceptionSpec represents the detail about JVM chaos action of Servlet throwing custom exceptions
+type ServletExceptionSpec struct {
+	// ServletCommonParameter represents the common servlet chaos parameter
+	ServletCommonParameter `json:",inline"`
+
+	// Exception represents the Exception class, with the full package name, must inherit from java.lang.Exception or Java.lang.Exception itself
+	Exception string `json:"exception"`
+
+	// Message represents specifies the exception class information.
+	// +optional
+	Message string `json:"message"`
+}
+
+// ServletDelaySpec represents the detail about JVM chaos action of Servlet response delay
+type ServletDelaySpec struct {
+	// ServletCommonParameter represents the common servlet chaos parameter
+	ServletCommonParameter `json:",inline"`
+
+	// Time represents delay time, in milliseconds, required
+	Time int `json:"time"`
+
+	// Offset represents delay fluctuation time
+	// +optional
+	Offset int `json:"offset"`
+}
+
+// ServletCommonParameter represents the common servlet chaos parameter
+type ServletCommonParameter struct {
+	// Method represents HTTP request method, such as GET, POST, or PUT. Default is GET
+	// +kubebuilder:validation:Enum=GET;POST;PUT
+	// +optional
+	Method string `json:"method"`
```

Comment:
What about other HTTP methods? I think they are also supported.
It seems we restrict enums in validation.

Reply:
I also think other HTTP methods can be supported, but I haven't verified them. In the Controller implementation, I'm going to validate them.

Reply:
I think at least DELETE is supported. Could we add it for now?